src/Illuminate/Support/Collection.php

```diff
@@ -768,6 +768,16 @@ public function toJson($options = 0)
 	}
 
 	/**
+	 * Get an hash of the collection.
+	 *
+	 * @return string
+	 */
+	public function hash()
+	{
+		return md5($this);
```

Comment:
`spl_object_hash()` will always return the same value for the same object.

`md5()` will convert the `Collection` to a string by calling `Collection::toJson()`.

So we relying on the JSON representation of the collection to be different to compare the collections. So any hashing algorithm will work fine, unless you have speed or memory requirements.